src/adhocracy_core/adhocracy_core/authentication/service_konto.py

```diff
@@ -0,0 +1,147 @@
+                      token: str) -> IUser:
+    """Authenticate ServiceKonto user.
+
+    Create the user if not yet existing. Update user data.
+    Returns:
+        user for ServiceKonto account
+    """
+    service_konto_enabled = asbool(registry.settings.get(
+        'adhocracy.service_konto.enabled', False))
+    if not service_konto_enabled:
+        raise ValueError('ServiceKonto authentication disabled.')
+    user_data_xml = _get_service_konto_user_data_xml(registry, token)
+    user_data = _parse_user_data_xml(user_data_xml)
+    user = _get_user(context, registry, request, int(user_data.get('userid')))
+    if not user:
+        user = _create_user(context, registry, request, user_data)
+    else:
+        _update_user(context, registry, request, user, user_data)
+    return user
+
+
+def _get_service_konto_user_data_xml(registry: Registry, token: str) -> str:
+    service_konto_url = registry.settings.get(
+        'adhocracy.service_konto.api_url')
+    try:
+        service_konto = osa.Client(service_konto_url)
+    except AttributeError:
+        raise ValueError('Failed connecting to ServiceKonto.')
+    result = service_konto.service.GetUserData(token)
+    if not result.GetUserDataResult == SERVICE_KONTO_GET_USER_DATA_SUCCESS:
+        raise ValueError('Failed getting user data (return_code: {}).'
+                         .format(result.GetUserDataResult))
+    return result.strXMLUserData
+
+
+def _parse_user_data_xml(xml: str) -> dict:
+    root = etree.fromstring(xml)
+    hhgw = _get_xml_element(root, 'HHGW')
+    if hhgw is None:
+        raise ValueError('Invalid data received.')
+    user_data = dict()
+    required_attributes = ['USERID', 'FIRSTNAME', 'LASTNAME', 'EMAIL']
+    for attribute in required_attributes:
+        user_data[attribute.lower()] = hhgw.get(attribute)
+    if not _is_user_data_valid(user_data):
+        raise ValueError('Invalid user data received.')
+    return user_data
+
+
+def _get_xml_element(root: etree.ElementTree, tag: str) -> etree.Element:
+    for element in root:
+        if element.tag == tag:
+            return element
+    return None
+
+
+def _is_user_data_valid(user_data: dict) -> bool:
+    is_valid = (user_data.get('userid') and
+                user_data.get('firstname') and
+                user_data.get('lastname') and
+                user_data.get('email'))
+    return is_valid
+
+
+def _get_user(context: IResource, registry: Registry, request: Request,
+              userid: int) -> IUser:
+    locator = registry.getMultiAdapter((context, request), IUserLocator)
+    user = locator.get_user_by_service_konto_userid(userid)
+    return user
+
+
+def _create_user(context: IResource, registry: Registry, request: Request,
+                 user_data: dict) -> IUser:
+    if _is_email_used(context, registry, request, user_data.get('email')):
+        raise ValueError('User with ServiceKonto email already exists.')
```

Comment:
To get things right: this error is thrown when a user has attempted to log in to meinberlin with a Servicekonto account whose email address is already used in a3, correct? If so, could you make the error message more descriptive?